wordpress-mobile/WordPress-Android
# Comment snippet: tracking

Part of #15660

This PR:
- adds source property for:
   - `comment_follow_conversation` (every follow/unfollow and even push action)
   🔵 Tracked: comment_follow_conversation, Properties: { "source":"reader_post_details_comments" }
   🔵 Tracked: comment_follow_conversation, Properties: { "source":"reader_threaded_comments" }
   - `reader_article_comments_opened`:
       - On a Reader post card, tap the comment button: 
       🔵 Tracked: reader_article_comments_opened, Properties: {"source":"reader_post_card"}     
       - In Reader post details, tap the comment button on the bottom toolbar.
       🔵 Tracked: reader_article_comments_opened, Properties: {"source":"reader_post_details"}
       - In Reader post details, tap the button in the Comments snippet
       🔵 Tracked: reader_article_comments_opened, Properties: {"source":"reader_post_details_comments"}
       - In Notifications > Comments, tap a notification's header that displays threaded comments.
       🔵 Tracked: reader_article_comments_opened, Properties: {"source":"comment_notification"}       
       - For a comment go to Notifications > Likes > detail > notification header. 
       🔵 Tracked: reader_article_comments_opened, Properties: {"source":"comment_like_notification"}
       - Go to My Site > Activity Log > Comment activity > tap the word Comment.
       🔵 Tracked: reader_article_comments_opened, Properties: {"source":"reader_post_details_comments"}
       - Open with the app a link to like a comment from an email notification
       🔵 Tracked: reader_article_comments_opened, Properties: {"source":"direct_operation"}
       
- adds a bit of unit testing

NOTE 1: the MY_SITE_COMMENT is currently not used but we  could want to mimic iOS here ([1](https://github.com/wordpress-mobile/WordPress-iOS/pull/17638), [2](https://github.com/wordpress-mobile/WordPress-iOS/pull/17651))
NOTE 2: if the comment thread is not in the cache yet you get a webview

## To test
- Check the unit tests are green
- Check you get the tracking as per description above

## Regression Notes
1. Potential unintended areas of impact
None

2. What I did to test those areas of impact (or what existing automated tests I relied on)
NA

3. What automated tests I added (or what prevented me from doing so)
Added a few unit test.

PR submission checklist:
